### DISCUSSION TITLE
Declare a common JS function for all page once instead of adding code at multiple needed points?

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8855] [Help] [answered]
I want to call the same JS function at multiple points inside the same page or even from different pages if possible, instead of copy pasting the same JS code at points where I need it. Is there any way to have this common at an embed component or at state to call from all these points? 
I have done a few tests but not successful so far.
For example for CSS there is a way to have a page affecting CSS code in an embed object. Isn't there a way to do something similar for a JS function?

**Comment by melohagan** [comment, marked as answer]
Hey @gmentzik 

I don't know of a way to do this at the moment. There's an issue you can follow here: https://github.com/Budibase/budibase/issues/3571

I don't see anything on the roadmap for the near term though.

**Comment by Simoon-F** [comment]
I need it！
